### PR TITLE
Roulette: Bonus item reward

### PR DESCRIPTION
**Pull request by Balferian**
* **Addressed Issue(s)**: -
* **Server Mode**: Pre/Re
* **Description of Pull Request**: 
Added bonus item reward for Lucky roulette.
1) Bonus item generated only once per game.
2) If reward item id the same as bonus item id then player get multiple (x2) amount.
3) If player start to play from mid or high stage then bonus item choosed only from current or higher stage.

Different items:
image
Same items:

### CODE REVIEW

File: src/map/battle.cpp [hunk 1]
```diff
@@ -10170,6 +10170,7 @@ static const struct _battle_data {
 	{ "homunculus_evo_intimacy_reset",      &battle_config.homunculus_evo_intimacy_reset,   1000,   0,      INT_MAX,        },
 	{ "monster_loot_search_type",           &battle_config.monster_loot_search_type,        1,      0,      1,              },
 	{ "feature.roulette",                   &battle_config.feature_roulette,                1,      0,      1,              },
+	{ "feature.roulette_bonus_reward",      &battle_config.feature_roulette_bonus_reward,   1,      0,      1,              },
```
[the remark this answers is not in the document]
Comment (Balferian) [reply src/map/battle.cpp:10173]: I forgot to move it back after tests. Thanks!

**Comment by MarkSucker** [comment]
Bonus item it's just still 1...
Do you check your code before approve?

**Comment by GravityClient** [comment]
After update [Update clif.cpp](https://github.com/rathena/rathena/pull/7427/commits/cf91a056e63221b7a423f3ead684bd73b4ceca25)

Lost bonus
Screenshot_6

**Comment by GravityClient** [comment]
Confirm it's works
Sorry i put it wrong line

Screenshot_7

**Comment by Lemongrass3110** [comment]
Basically everything works now!
I am just not sure if this logic is correct. But we cant really check on official servers anymore.
Based on the packets I would guess:
- If you open the roulette window the bonus item should already be calculated (see clif_roulette_open packet)
- If you retrieve the prize the next bonus item should already be calculated (see clif_roulette_recvitem_ack packet)

I think this is so that the player can choose if the bonus item is worth his points or not maybe?

**Comment by Balferian** [comment]
> I think this is so that the player can choose if the bonus item is worth his points or not maybe?

I was looking again video from Ziu.
When he open roulette bonus item already selected.
I can`t understand the logic when bonus item should be changed.
https://youtu.be/u-_45Addgh0
https://youtu.be/n76720cG1ac

File: conf/battle/feature.conf [hunk 2]
```diff
@@ -67,6 +67,10 @@ feature.bgqueue: on
 // Requires: 2014-10-22bRagexe or later
 feature.roulette: on
 
+// Roulette bonus reward
+// Multiply amount by 2 if the reward item id is the same as bonus item id
```
Comment (aleos89) [inline conf/battle/feature.conf:71]:
```suggestion
// Multiply amount by 2 if the reward item ID is the same as bonus item ID
```

**Review by aleos89 — APPROVED** [review]

**Pull request merged**